ed_sensor_integration/src/laser/updater.cpp

```diff
@@ -0,0 +1,655 @@
+                continue;
+
+            entities.push_back(e);
+        }
+    }
+    return entities;
+}
+
+/**
+ * @brief associateSegmentsWithEntities
+ * @param[in] clusters measured clusters
+ * @param[in] entities entities that may be associated with the clusters
+ * @param[in] current_time current time to compare against the last measurement of an entity
+ * @param[out] assig assignmentmatrix between clusters and entities.
+ * @return whether or not the assignment was successful
+ */
+bool associateSegmentsWithEntities(std::vector<EntityUpdate>& clusters, const  std::vector<ed::EntityConstPtr>& entities, double current_time, ed_sensor_integration::Assignment& assig)
+{
+    // Create association matrix
+    ed_sensor_integration::AssociationMatrix assoc_matrix(clusters.size());
+    for (unsigned int i_cluster = 0; i_cluster < clusters.size(); ++i_cluster)
+    {
+        const EntityUpdate& cluster = clusters[i_cluster];
+
+        for (unsigned int i_entity = 0; i_entity < entities.size(); ++i_entity)
+        {
+            const ed::EntityConstPtr& e = entities[i_entity];
+
+            const geo::Pose3D& entity_pose = e->pose();
+            const ed::ConvexHull& entity_chull = e->convexHull();
+
+            float dx = entity_pose.t.x - cluster.pose.t.x;
+            float dy = entity_pose.t.y - cluster.pose.t.y;
+            float dz = 0;
+
+            if (entity_chull.z_max + entity_pose.t.z < cluster.chull.z_min + cluster.pose.t.z
+                    || cluster.chull.z_max + cluster.pose.t.z < entity_chull.z_min + entity_pose.t.z)
+                // The convex hulls are non-overlapping in z
+                dz = entity_pose.t.z - cluster.pose.t.z;
+
+            float dist_sq = (dx * dx + dy * dy + dz * dz);
+
+            // TODO: better prob calculation
+            double prob = 1.0 / (1.0 + 100 * dist_sq);
+
+            double dt = current_time - e->lastUpdateTimestamp();
+
+            double e_max_dist = std::max(0.2, std::min(0.5, dt * 10));
+
+            if (dist_sq > e_max_dist * e_max_dist)
+                prob = 0;
+
+            if (prob > 0)
+                assoc_matrix.setEntry(i_cluster, i_entity, prob);
+        }
+    }
+    return assoc_matrix.calculateBestAssignment(assig);
+}
+
+LaserUpdater::LaserUpdater()
+{
+}
+
+LaserUpdater::~LaserUpdater()
+{
+}
+
+void LaserUpdater::configure(tue::Configuration& config)
+{
+    config.value("world_association_distance", world_association_distance_);
+    config.value("min_segment_size_pixels", min_segment_size_pixels_);
+    config.value("segment_depth_threshold", segment_depth_threshold_);
+    config.value("min_cluster_size", min_cluster_size_);
+    config.value("max_cluster_size", max_cluster_size_);
+    config.value("max_gap_size", max_gap_size_);
+
+    int i_fit_entities = 0;
+    config.value("fit_entities", i_fit_entities, tue::config::OPTIONAL);
+    fit_entities_ = (i_fit_entities != 0);
```

Comment:
Looks like a bool to me.

Reply:
fit_entities_ is a bool. But I did not want to change the interface of the plugin. That is another task for another day